### PR TITLE
feat: set ProbeTerminationGracePeriod default to true

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind feature
/sig node

#### What this PR does / why we need it:
As mentioned on https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2238-liveness-probe-grace-period#beta-122

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/2238

#### Does this PR introduce a user-facing change?

```release-note
Feature gate `ProbeTerminationGracePeriod` is enabled by default.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/2238
```

### CODE REVIEW

**Review by ehashman — COMMENTED** [review]
/lgtm
/approve
/triage accepted
/priority important-soon
for 1.25

Will quickly update KEP metadata for 1.25 so this can be included. Leaving hold on until then.

**Comment by marosset** [comment]
Please update https://github.com/kubernetes/kubernetes/blob/2b51b2595ad63e59dcbab1f95620ed0ecc8c6d8c/pkg/features/kube_features.go#L630-L634 

Also, this should have a release note.

**Comment by kerthcet** [comment]
> Please update
> 
> https://github.com/kubernetes/kubernetes/blob/2b51b2595ad63e59dcbab1f95620ed0ecc8c6d8c/pkg/features/kube_features.go#L630-L634
> 
> Also, this should have a release note.

Good catch, thanks.

**Comment by marosset** [comment]
> > Please update
> > https://github.com/kubernetes/kubernetes/blob/2b51b2595ad63e59dcbab1f95620ed0ecc8c6d8c/pkg/features/kube_features.go#L630-L634
> > 
> > Also, this should have a release note.
> 
> Good catch, thanks.

Thanks!

**Comment by kerthcet** [comment]
since https://github.com/kubernetes/enhancements/pull/3282 merged
/hold cancel 

can you add the LGTM again @dchen1107

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -652,6 +652,7 @@ const (
 
 	// owner: @ehashman
 	// alpha: v1.21
+	// beta: v1.22
```
Comment (psschwei) [inline pkg/features/kube_features.go:655]:
```suggestion
	// beta: v1.25
```

Comment (kerthcet) [reply pkg/features/kube_features.go:655]: It actually graduated to beta in 1.22. We just set the default value to true in 1.25.

Comment (psschwei) [reply pkg/features/kube_features.go:655]: got it... so maybe `// beta: v1.22 (off by default until 1.25)` ? as per convention in https://github.com/kubernetes/kubernetes/blame/master/pkg/features/kube_features.go#L258

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -986,7 +987,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	PreferNominatedNode: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.26
 
-	ProbeTerminationGracePeriod: {Default: false, PreRelease: featuregate.Beta}, // Default to false in beta 1.22, set to true in 1.24
+	ProbeTerminationGracePeriod: {Default: true, PreRelease: featuregate.Beta}, // Default to true in beta 1.25
```
Comment (aojea) [inline pkg/features/kube_features.go:990]: @liggitt I think that this is not covered in https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/3136-beta-apis-off-by-default

Comment (aojea) [reply pkg/features/kube_features.go:990]: It is a question, this was beta before the kep, it should be able to transition to enabled by default, no?

Comment (liggitt) [reply pkg/features/kube_features.go:990]: yes, that KEP is only referring to beta endpoints / API versions

Comment (liggitt) [reply pkg/features/kube_features.go:990]: (specifically called out in the non-goals - https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/3136-beta-apis-off-by-default#non-goals)

> **Non-Goals**
>
> Change feature gate defaults. Feature gates control new features (not just new APIs) and they are on by default for beta features. This KEP is not changing the lifecycle flow for feature gates. It is currently alpha=off-by-default, beta=on-by-default, stable=locked-to-on.

Comment (kerthcet) [reply pkg/features/kube_features.go:990]: [3136-beta-apis-off-by-default](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/3136-beta-apis-off-by-default) only focus on the beta APIs, we can still change the default value of feature gates here, right? So there's no problem here.

File: pkg/features/kube_features.go (hunk 1 above)
Comment (kerthcet) [reply pkg/features/kube_features.go:655]: We covered this description below in L990.

File: pkg/features/kube_features.go (hunk 2 above)
Comment (aojea) [reply pkg/features/kube_features.go:990]: I was mixing endpoints and feature gates, forgive me

**Comment by mrunalp** [comment]
@psschwei ptal

**Comment by psschwei** [comment]
Question: in the [updated test plan for this KEP](https://github.com/kubernetes/enhancements/tree/6270ce1966a279bf52d7aea9bc2576651f41dbad/keps/sig-node/2238-liveness-probe-grace-period#test-plan), we said we would add some unit and integration buckets. Do those need to be added before we flip the gate? 

(for reference, #110733 adds the units, and I plan to get the integration ones in next week)

**Comment by kerthcet** [comment]
cc @dchen1107 since we're close to the code freeze.

**Comment by kerthcet** [comment]
cc @endocrimes @rphillips

**Comment by endocrimes** [comment]
/lgtm
/hold

E2E coverage for this exists iirc (please confirm) so I don't have much in the way of test qualms with this progressing.

Lgtm with hold to let the unit coverage land first.

**Comment by dchen1107** [comment]
Marked this for 1.25 milestone explicitly, but please point us to the e2e tests. Thanks!

**Comment by psschwei** [comment]
links to e2e tests:
* https://github.com/kubernetes/kubernetes/blob/262d8b2a77ef1e25d8968fc6bcd56a0c5b8f8f90/test/e2e/common/node/container_probe.go#L459
* https://github.com/kubernetes/kubernetes/blob/262d8b2a77ef1e25d8968fc6bcd56a0c5b8f8f90/test/e2e/common/node/container_probe.go#L487

**Comment by mrunalp** [comment]
The unit tests PR has landed #110733. We can let this go in.

**Pull request merged**